libcasr/src/gdb/exploitable.rs

```diff
@@ -436,6 +441,76 @@ impl GdbContext {
         }
         ExecutionClass::find("AccessViolation")
     }
+
+    /// Analyze aarch64 crash instruction
+    ///
+    /// # Arguments
+    ///
+    /// * `cs` - reference to capstone.
+    ///
+    /// * `insns` - reference to disassembled instructions.
+    ///
+    /// * `info` - reference to signal information struct.
+    fn analyze_instructions_arm64(
+        cs: &Capstone,
+        insns: &Instructions,
+        info: &Siginfo,
+    ) -> Result<ExecutionClass> {
+        // Get first instruction.
+        let Some(insn) = insns.iter().next() else {
+            return Err(Error::Casr(
+                "Couldn't get first arm instruction".to_string(),
+            ));
+        };
+
+        let Ok(detail) = cs.insn_detail(insn) else {
+            return Err(Error::Casr(
+                "Couldn't capstone instruction details".to_string(),
+            ));
+        };
+
+        // Check for mov instructions.
+        let Some(mnemonic) = insn.mnemonic() else {
+            return Err(Error::Casr(
+                "Couldn't capstone instruction mnemonic".to_string(),
+            ));
+        };
+        let m = mnemonic.to_string();
+
+        let ops = detail.arch_detail().operands();
+        for op in ops.iter() {
+            // Safe.
+            let capstone::arch::ArchOperand::Arm64Operand(operand) = op else {
+                return Err(Error::Casr(
+                    "Couldn't capstone instruction operands".to_string(),
+                ));
+            };
+            // Check mem operand.
+            if let capstone::arch::arm64::Arm64OperandType::Mem(_) = operand.op_type {
+                match (
+                    info.si_code,
+                    m.contains("st"),
+                    m.contains("ld"),
```

Comment:
Is this actually selects only loads and stores?

Reply:
Let's use mnemonics from check_taint for load and store. Maybe we could create a functions is_load_arm64, is_store_arm64.

Reply:
done, but mnemonics for load and store in exploitable.rs do not cover all instructions that work with memory. Should we add entire set of memory instruction?

Reply:
Any examples? Only instructions that load/store registers are interested and supported by capstone also.

Reply:
STSET, CASA, LDAR?

Reply:
STSET - is bit set (suppose no), CAS -compare and swap (suppose no), LDAR - maybe, If you think it's necessary you could and it.